src/PlusDataCollection/PointGrey/vtkPlusSpinnakerVideoSource.cxx

```diff
@@ -0,0 +1,807 @@
+      this->Internal->CameraPtr->Gain.SetValue(this->GainDB);
+    }
+    else if (this->GainMode == GAIN_AUTO_ONCE)
+    {
+      this->Internal->CameraPtr->GainAuto.SetValue(Spinnaker::GainAuto_Once);
+    }
+    else if (this->GainMode == GAIN_AUTO_CONTINUOUS)
+    {
+      this->Internal->CameraPtr->GainAuto.SetValue(Spinnaker::GainAuto_Continuous);
+    }
+
+    // set white balance mode && wb (if manual wb control enabled)
+    if (this->WhiteBalanceMode == WB_MANUAL)
+    {
+      this->Internal->CameraPtr->BalanceWhiteAuto.SetValue(Spinnaker::BalanceWhiteAutoEnums::BalanceWhiteAuto_Off);
+      this->Internal->CameraPtr->BalanceRatioSelector.SetValue(Spinnaker::BalanceRatioSelectorEnums::BalanceRatioSelector_Blue);
+      Spinnaker::GenApi::CFloatPtr balanceRatioPtr = nodeMap.GetNode("BalanceRatio");
+      balanceRatioPtr->SetValue(this->WhiteBalanceBlue);
+      this->Internal->CameraPtr->BalanceRatioSelector.SetValue(Spinnaker::BalanceRatioSelectorEnums::BalanceRatioSelector_Red);
+      balanceRatioPtr->SetValue(this->WhiteBalanceRed);
+    }
+    else if (this->WhiteBalanceMode == WB_AUTO_ONCE)
+    {
+      this->Internal->CameraPtr->BalanceWhiteAuto.SetValue(Spinnaker::BalanceWhiteAutoEnums::BalanceWhiteAuto_Once);
+    }
+    else if (this->WhiteBalanceMode == WB_AUTO_CONTINUOUS)
+    {
+      this->Internal->CameraPtr->BalanceWhiteAuto.SetValue(Spinnaker::BalanceWhiteAutoEnums::BalanceWhiteAuto_Continuous);
+    }
+
+    // set sharpening mode && sharpening (if manual sharpening control enabled)
+    //if (this->SharpeningMode == SHARPENING_MANUAL)
+    //{
+    //  this->Internal->CameraPtr->SharpeningEnable.SetValue(true);
+    //  this->Internal->CameraPtr->SharpeningAuto.SetValue(false);
+    //  this->Internal->CameraPtr->Sharpening.SetValue(this->SharpeningAmount);
+    //}
+    //else if (this->SharpeningMode == SHARPENING_AUTO)
+    //{
+    //  this->Internal->CameraPtr->SharpeningEnable.SetValue(true);
+    //  this->Internal->CameraPtr->SharpeningAuto.SetValue(true);
+    //}
+    //else if (this->SharpeningMode == SHARPENING_OFF)
+    //{
+    //  this->Internal->CameraPtr->SharpeningEnable.SetValue(false);
+    //}
+  }
+  catch (Spinnaker::Exception &e)
+  {
+    LOG_ERROR("SpinnakerVideoSource: Error configuring camera. Exception text: " << e.what());
+    return PLUS_FAIL;
+  }
+
+  return PLUS_SUCCESS;
+}
+
+//----------------------------------------------------------------------------
+PlusStatus vtkPlusSpinnakerVideoSource::InternalDisconnect()
+{
+  LOG_TRACE("vtkPlusSpinnakerVideoSource::InternalDisconnect()");
+
+  try
+  {
+    this->Internal->CameraPtr->DeInit();
+    this->Internal->CameraPtr = NULL;
+    this->Internal->CameraList.Clear();
+    this->Internal->SystemPtr->ReleaseInstance();
+  }
+  catch (Spinnaker::Exception &e)
+  {
+    LOG_ERROR("SpinnakerVideoSource: Failed to de-initialize camera. Exception text: " << e.what());
```

Comment:
This needs to guarantee as best as possible that memory has been deallocated. Could you make the try catch block more granular so you can more accurately detect what failed and compensate in each catch block?